apis/Google.Cloud.DevTools.Common/Google.Cloud.DevTools.Common/Source/V1/SourceContextFile.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using System;
+using System.IO;
+
+namespace Google.Cloud.DevTools.Source.V1
+{
+    /// <summary>
+    /// Utilities that reads source-context.json file from app root.
+    /// </summary>
+    public static class SourceContextFile
+    {
+        private const string SourceContextFileName = "source-context.json";
+        private static Lazy<string> s_filePath = new Lazy<string>(GetFilePath);
+        private static Lazy<SourceContext> s_sourceContext = new Lazy<SourceContext>(OpenParseFile);
+        private static Lazy<string> s_gitRevisionId => new Lazy<string>(() => s_sourceContext.Value?.Git.RevisionId);
```

Comment:
`Git?.RevisionId`, so that if there's a context without a git part, we just get null instead of an exception?